src/store/index.js

```diff
@@ -5,21 +5,36 @@
  * @flow
  */
 
+import Rfc822Address from '../utility/Rfc822Address';
+
 import type { Data, Message } from './types';
 
 const store: Data = require('./emails.json');
 
 /**
- * You might find it useful to implement some functions that make accessing
- * deeply nested attributes of Messages or Threads...
+ * Message helpers
  */
 
-// For example, this returns the subject of a given message if it is defined
-// by a header in the message payload
+export function getMessageSender(message: Message): Rfc822Address {
+  const { headers } = message.payload;
+  const fromHeader = headers.find(header => header.name === 'From');
+
+  if (fromHeader == null) {
+    throw new Error('Every email must have been sent by someone!');
+  }
```

Comment:
I used the [React 16 Idea](https://reactjs.org/docs/error-boundaries.html#new-behavior-for-uncaught-errors) you shared with me to simply not render a list item for an email with no sender since in theory this shouldn't happen...

I don't see where you are handling the exception where you are using this fn though, so in theory, wouldn't that crash your app?

Reply:
Thanks for the review, @mariehooper. You're right, this error is unhandled. I’m not following my own advice; just punting on error handling at this point. 😅

For now, with the JSON data store, this code will definitely not run, so I’m gonna call this outside the scope of this PR. I’ll have to deal with this when we’re getting the emails from an API, but it’s hard to decide what the “right” way to fail would be, since an email without a sender could only be a symptom of a larger problem. Certainly hitting a reporting service would be a good start.